src/main/java/com/chikli/hudson/plugin/naginator/NaginatorListener.java

```diff
@@ -27,92 +21,53 @@
 
 
     @Override
-    public void onCompleted(AbstractBuild<?, ?> build, TaskListener listener) {
-        if ((build.getResult() == SUCCESS) || (build.getResult() == ABORTED)) {
-            return;
-        }
-
-        NaginatorPublisher naginator = build.getProject().getPublishersList().get(NaginatorPublisher.class);
-
-        // JENKINS-13791
-        if (naginator == null) {
-            return;
-        }
-
-        // If we're not set to rerun if unstable, and the build's unstable, return true.
-        if ((!naginator.isRerunIfUnstable()) && (build.getResult() == Result.UNSTABLE)) {
-            return;
-        }
-
+    public void onCompleted(AbstractBuild<?, ?> build, @Nonnull TaskListener listener) {
         // Do nothing for a single Matrix run. (Run only when all Matrix finishes)
-        if (build instanceof MatrixRun) {
+        if (build == null || build instanceof MatrixRun) {
```

Comment:
`null instanceof SomeClass` always returns `false`: http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2
(though, that sounds a little strange to me as `MatrixRun` can be `null`...)
This null-check is not for `instanceof` but for following codes (e.g. `calculateRetryCount`).

The code comment above may be confusing and I'll update that.